vignettes/design-principles.Rmd

```diff
@@ -0,0 +1,48 @@
+---
+title: "Design Principles for {packagetemplate}"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Design Principles for {packagetemplate}}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette outlines the design decisions that have been taken during the development of the {packagetemplate} R package, and provides some of the reasoning, and possible pros and cons of each decision.
+
+This document is primarily intended to be read by those interested in understanding the code within the package and for potential package contributors.
+
+<!-- Text in angular brackets (<>) is to guide what to include for each section please delete the guidelines and replace for your package.
+
+None of the sections are required, feel free to remove any sections not relevant to your package or add new sections if helpful. -->
+
+## Scope
+
+< Outline the aims of the package, potentially mention some of the key exported functions, and maybe how it links with other R packages. It is also possible to mention certain aspects that fall outside of the package's scope. >
+
+## Input/Output/Interoperability
+
+< Describe the data structures (i.e. vectors, `<data.frames>` or classes) that are given as input to the key functions and what data structures the functions return. The design decisions around these I/O choices could also mention how it enhances interoperability with other R packages or pipelines (e.g. with `%>%`). >
+
+
+## Design decisions
+
+< A list of bullet points each explaining a design decision and its reasoning. >
+
+## Dependencies
+
+< A list of dependencies used by the package with some explanation as to why they are required. Not all dependencies need to be explained. This section can also mention dependencies that are planned to be removed or added in future development. >
+
+## Development journey
+
+<  If the package has undergone any large refactoring this section can be used to explain the changes. >
+
+## Contribute
+
+< Any additional information for a developer to help them contribute to the package, do not repeat information from the [package contributing guide](https://github.com/epiverse-trace/.github/blob/main/CONTRIBUTING.md). >
```

Comment:
Do you have a specific situation in mind where it would be more appropriate to have this info here rather than in a custom contributing guide?

Reply:
Yes, can you submit a follow up PR removing this please? I'm afraid it will encourage the addition of info that should be in the contributing guide instead.